src/tweenable.js

```diff
@@ -323,8 +323,13 @@ const remove = ((previousTween, nextTween) => tween => {
 })()
 
 const defaultPromiseCtor = typeof Promise === 'function' ? Promise : null
-
+/**
+ * @class
+ * @implements {Promise<Object>}
+ */
```

Comment:
Should this probably be `object` or `unknown` as any is pretty dangerous.  But I don't want to break anybodies typing

Reply:
That seems reasonable. I think we may be stuck with `Promise<Object>` here until there's a full conversion to TypeScript. With pure TypeScript, we could use a class generic to define what the `Promise` returns, but I don't know that the current JSDoc-based type implementation would support that. Perhaps @arthuro555  might have some insight here?

Reply:
I agree that `unknown` would be a good idea as the default, even after adding generics.

We could definitely use a generic to define what is to be returned by the promise once we move to typescript. JSDoc should have a syntax for generics too, but when I tried to use it there was a problem (i forgot what it was exactly) that made them not usuable in this context.

Reply:
Thanks for weighing in @arthuro555! Would there be an advantage to using `unknown` instead of `Object` here? `Object` seems a bit more intuitive to me, but aside from that I don't have a strong case for using that instead of `unknown`.

Reply:
`unknown` tells the user and typescript that the library does not know the returned type, and that if you want to use the returned value you will have to either cast it to the correct type or narrow it down. That forces users of the library to type their parameter correctly. Otherwise, they'll see no error when accessing wrong properties and assume they made nothing wrong since no error will pop-up. To ensure type safety is kept at all time, it is therefore imo better to use unknown here.

Reply:
That makes a lot of sense. Thank you for the excellent explanation @arthuro555!

@jspears, would you mind changing the `@implements` type to `Promise<unknown>`?